internal/interceptor.go

```diff
@@ -231,6 +231,9 @@ type WorkflowOutboundInterceptor interface {
 	// NewTimer intercepts workflow.NewTimer.
 	NewTimer(ctx Context, d time.Duration) Future
 
+	// NewTimer intercepts workflow.NewTimerWithOptions.
+	NewTimerWithOptions(ctx Context, d time.Duration, options TimerOptions) Future
+
```

Comment:
Often the SDK will create internal timers, it would be nice if we can leverage metadata to give those timers appropriate names. Not saying you need to do that in this PR, but if we agree we should open an issue

Reply:
Agreed. Will set reminder to open issue there.

Reply:
Opened #1602